gossip/state/state.go

```diff
@@ -758,6 +784,9 @@ func (s *GossipStateProviderImpl) filterPeers(predicate func(peer discovery.Netw
 	var peers []*comm.RemotePeer
 
 	for _, member := range s.mediator.PeersOfChannel(common2.ChainID(s.chainID)) {
+		if !s.CanPredicate(member) {
```

Comment:
Instead of CanPredicate, you can pass in another predicate function which overrides the behaviour.

Reply:
sounds good, will do that

Reply:
done